src/uiveri5.js

```diff
@@ -270,31 +270,51 @@ function run(config) {
 
       });
 
-      // override with added logging and parameter manipulation
+      // used in protractor for many scripts and notably for waitForAngular
+      // override with added logging
       var origExecuteAsyncScript_= browser.executeAsyncScript_;
       browser.executeAsyncScript_ = function() {
+        var code = arguments[0];
+        var scriptName = arguments[1];
+        var params = arguments[2];
         // log script execution
-        logger.trace('Execute async script: ${name}, code:\n ${JSON.stringify(code)}',
-          {name:  arguments[1], code: arguments[0]});
-        // override the timeout used by waitForAngular
-        arguments[2] = JSON.stringify({
-          waitForUI5Timeout: waitForUI5Timeout
-        });
+        logger.trace('Execute async script: ' + scriptName + ' with params: ' + JSON.stringify(params) + ' with code: \n' + JSON.stringify(code));
         //call original function in its context
-        return origExecuteAsyncScript_.apply(browser, arguments);
+        return origExecuteAsyncScript_.apply(browser, arguments)
+          .then(function(res) {
+            logger.trace('Async script: ' + scriptName + ' result: ' + JSON.stringify(res));
+            return res;
+          },function(error) {
+            logger.trace('Async script: ' + scriptName + ' error: ' + JSON.stringify(error));
+            throw error;
+          })
+      };
+
+      browser.executeAsyncScriptHandleErrors = function executeAsyncScriptLogErrors(scriptName,params) {
+        var code = clientsidescripts[scriptName];
```

Comment:
itsnt it too limiting to only work with functions defined in clientsidescripts?  (eg first arg could also be a function)

Reply:
I would like to limit this practice as it makes it harder to comply with a common script structure